docs/admin/out-of-resource.md

```diff
@@ -0,0 +1,220 @@
+| memory.available | memory.available := node.status.capacity[memory] - node.stats.memory.workingSet |
+
+In future releases, the `kubelet` will support the ability to trigger eviction decisions based on disk pressure.
+
+Until that time, it is recommended users take advantage of [garbage collection](/docs/admin/garbage-collection/).
+
+### Eviction Thresholds
+
+The kubelet supports the ability to specify eviction thresholds that trigger the kubelet to reclaim resources.
+
+Each threshold is of the following form:
+
+`<eviction-signal><operator><quantity>`
+
+* valid `eviction-signal` tokens as defined above.
+* valid `operator` tokens are `<`
+* valid `quantity` tokens must match the quantity representation used by Kubernetes
+
+#### Soft Eviction Thresholds
+
+A soft eviction threshold pairs an eviction threshold with a required
+administrator specified grace period.  No action is taken by the `kubelet`
+to reclaim resources associated with the eviction signal until that grace
+period has been exceeded.  If no grace period is provided, the `kubelet` will
+error on startup.
+
+In addition, if a soft eviction threshold has been met, an operator can
+specify a maximum allowed pod termination grace period to use when evicting
+pods from the node.  If specified, the `kubelet` will use the lesser value among
+the `pod.Spec.TerminationGracePeriodSeconds` and the max allowed grace period.
+If not specified, the `kubelet` will kill pods immediately with no graceful
+termination.
+
+To configure soft eviction thresholds, the following flags are be supported:
+
+```
+--eviction-soft="": A set of eviction thresholds (e.g. memory.available<1.5Gi) that if met over a corresponding grace period would trigger a pod eviction.
+--eviction-soft-grace-period="": A set of eviction grace periods (e.g. memory.available=1m30s) that correspond to how long a soft eviction threshold must hold before triggering a pod eviction.
+--eviction-max-pod-grace-period="0": Maximum allowed grace period (in seconds) to use when terminating pods in response to a soft eviction threshold being met.
+```
+
+#### Hard Eviction Thresholds
+
+A hard eviction threshold has no grace period, and if observed, the `kubelet`
+will take immediate action to reclaim the associated starved resource.  If a
+hard eviction threshold is met, the `kubelet` will kill the pod immediately
+with no graceful termination.
+
+To configure hard eviction thresholds, the following flag is supported:
+
+```
+--eviction-hard="": A set of eviction thresholds (e.g. memory.available<1Gi) that if met would trigger a pod eviction.
+```
+
+### Eviction Monitoring Interval
+
+The `kubelet` evaluates eviction thresholds per its configured housekeeping inteval.
+
+```
+--housekeeping-interval=10s: Interval between container housekeepings
+```
+
+### Node Conditions
+
+The `kubelet` will support a node condition that corresponds to each eviction signal.
```

Comment:
I didn't mean to imply that each signal gets a unique condition.  Just that a mapping from signal to some condition exists.